main.js

```diff
@@ -0,0 +1,173 @@
+    alert('Please fill in both fields.');
+    return;
+  };
+
+  // text from the input added to the post container
+  const newPost = document.createElement('div');
+  const newPostTextEl = document.createElement('p');
+  const newPostTextNode = document.createTextNode(postText);
+  newPostTextEl.appendChild(newPostTextNode);
+
+  // name from the input added to the post container
+  const newPostNameEl = document.createElement('p');
+  const newPostNameNode = document.createTextNode('- Posted By: ' + postName);
+
+  // make the name element bold and appending
+  newPostNameEl.style.fontWeight = 'bold';
+  newPostNameEl.appendChild(newPostNameNode);
+
+  // create a break line between posts
+  const divider = document.createElement('hr');
+  
+  // create delete button and add event listener to it
+  const deleteBtn = document.createElement('button');
+  deleteBtn.setAttribute('id', 'delete-btn');
+  deleteBtn.setAttribute('type', 'button');
+  deleteBtn.setAttribute('class', 'btn btn-danger btn-sm fa-solid fa-delete-left pull-right delete-btn');
+  deleteBtn.addEventListener('click', (event) => {
+    event.target.parentElement.remove()
+  });
+  
+  // create a button that opens the comment form
+  const commentFormBtn = document.createElement('button');
+  commentFormBtn.setAttribute('class', 'comment-form btn btn-warning');
+  commentFormBtn.setAttribute('type', 'button');
+  commentFormBtn.innerHTML = 'Comments';
+  
+  // append post element, inputs, buttons into post div
+  newPost.append(newPostTextEl);
+  newPost.append(newPostNameEl);
+  newPost.append(commentFormBtn);
+  newPost.append(deleteBtn);
+  
+  // add the new post to the post container
+  postContainer.append(newPost);
+
+  // clear the input fields
+  textInput.value = '';
+  nameInput.value = '';
+
+  // create container for comments
+  const commentContainer = document.createElement('div');
+  commentContainer.setAttribute('class', 'comments-container');  
+  
+  // append commentContainer to newPost
+  newPost.append(commentContainer); 
+
+  // event listener for the comment form button. When the button is clicked, create the comment form
+  commentFormBtn.addEventListener('click', () => {
+  
+    // create the comment input form matching post form
+    const commentForm = document.createElement('form');
+    commentForm.setAttribute('style', 'margin-top: 30px');
+    commentForm.setAttribute('onsubmit', 'event.preventDefault();');
+    
+    // create the comment text element 
+    const commentTextForm = document.createElement('div');
+    commentTextForm.setAttribute('class', 'form-group');
+    
+    // create the comment textarea input field
+    const commentTextInput = document.createElement('textarea');
+    commentTextInput.setAttribute('id', 'comment-msg');
+    commentTextInput.setAttribute('type', 'text');
+    commentTextInput.setAttribute('class', 'form-control');
+    commentTextInput.setAttribute('placeholder', 'Write a comment');
```

Comment:
a lot of repetition. [DRY](https://docs.getdbt.com/terms/dry). 
You can create a function that takes the params. Or alternatively use your JS helper methods:
```suggestion
const atributes = [
   { name: 'id', value: 'comment-msg' },
   { name: 'type', value: 'text' },
//and the others
];
// helper to the rescue!
atributes.forEach((attribute) => {
   commentTextInput.setAttribute(name, value);
});
```
Will make your code considerable better!